framework/contrib/hit/lex.h

```diff
@@ -0,0 +1,151 @@
+#ifndef HIT_LEX
+#define HIT_LEX
+
+#include <string>
+#include <vector>
+
+namespace hit
+{
+
+// the EOF macro wreaks havok on our TokType enum which has an EOF member. So undefine it and the
+// redefine it at the end of the file.
+#undef EOF
+
+// TokType
+enum class TokType
+{
+  Error,
+  Equals,
+  LeftBracket,
+  RightBracket,
+  Ident,
+  Path,
+  Number,
+  String,
+  Comment,
+  InlineComment,
+  EOF,
+};
+
+/// Token represents an (atomic) token/quantum of input text.
+struct Token
+{
+  Token(TokType t, std::string val, size_t offset = 0, int line = 0);
+  /// str returns a human-friendly string representation of the token.
+  std::string str();
+
+  /// type identifies the category/type of the token (i.e. String, Number, Comment, etc.)
+  TokType type;
+  /// val is the actual text from the input that makes this token.
+  std::string val;
+  /// offset is the byte offset into the original input identifying the start position where this
+  /// token was found.  This can be used to determine line numbers, column offsets, etc. useful for
+  /// error messages among other things.
+  size_t offset;
+  /// line is the line number of the original input (lines deliminted by the unix newline '\n'
+  /// character) on which the beginning of the token was found - this is redundant with the offset
+  /// combined with a reference to the original input, but is here for convenience.
```

Comment:
we use 
```
/**
 *
 */
```

For multiline Doxygen comments

Reply:
I know this is contrib so it doesn't necessarily have to look the same as MOOSE, but the block style does look much nicer where you have these large text blocks.